quickwit/quickwit-indexing/src/source/queue_sources/shared_state.rs

```diff
@@ -125,7 +191,7 @@ impl QueueSharedState {
 /// Acquires shards from the shared state for the provided list of messages and
 /// maps results to that same list
 pub async fn checkpoint_messages(
```

Comment:
to be discussed: I don't understand what this function does, nor its naming.

Reply:
> also I am not fond of the iterator style with a non pure function

I got rid of those.